contributors/design-proposals/deployment.md

```diff
@@ -163,52 +180,61 @@ LastTransitionTime to PodCondition.
-  when the image:v2 RC had been ramped up to 5 pods and the image:v1 RC
+- User then updates this Deployment to create pods with image:v3,
+  when the image:v2 RS had been ramped up to 5 pods and the image:v1 RS
   had been ramped down to 5 pods.
-- When Deployment Controller observes the new deployment, it will create
-  a new RC for creating pods with image:v3. It will then start ramping up this
-  new RC to 10 pods and will ramp down both the existing RCs to 0.
+- When Deployment Controller observes the new update, it will create
+  a new RS for creating pods with image:v3. It will then start ramping up this
+  new RS to 10 pods and will ramp down both the existing RSs to 0.
 
 ### Deleting
 
-Users can pause/cancel a deployment by deleting it before it is completed.
-Recreating the same deployment will resume it.
-For ex: consider the following case:
-- User creates a deployment to rolling-update 10 pods with image:v1 to
+Users can cancel a rollout by doing a non-cascading deletion of the Deployment
+it before it is complete. Recreating the same Deployment will resume it.
+For example, consider the following case:
+- User creates a Deployment to rolling-update 10 pods with image:v1 to
   pods with image:v2.
-- User then deletes this deployment while the old and new RCs are at 5 replicas each.
-  User will end up with 2 RCs with 5 replicas each.
-User can then create the same deployment again in which case, DeploymentController will
-notice that the second RC exists already which it can ramp up while ramping down
+- User then deletes this Deployment while the old and new RSs are at 5 replicas each.
+  User will end up with 2 RSs with 5 replicas each.
+User can then create the same Deployment again in which case, DeploymentController will
+notice that the second RS exists already which it can ramp up while ramping down
 the first one.
 
 ### Rollback
 
-We want to allow the user to rollback a deployment. To rollback a
-completed (or ongoing) deployment, user can create (or update) a deployment with
-DeploymentSpec.PodTemplateSpec = oldRC.PodTemplateSpec.
+We want to allow the user to rollback a Deployment. To rollback a
+completed (or ongoing) Deployment, users can simply use `kubectl rollout undo`
+and specify the revision they want to rollback to or no revision which means
+that the Deployment will be rolled back to its previous revision.
+By hashing PodTemplateSpec, rollback moves the ReplicaSet a user rolls back to as
+the new ReplicaSet. The name for the new ReplicaSet will be always unique but
+predictable across versions of Kubernetes by using TemplateGeneration. That means
+we cannot reuse old ReplicaSets as before TemplateGeneration - the controller
+should do a non-cascading deletion of the old ReplicaSet that would orphan any
+existing Pods and create the new ReplicaSet that will adopt any existing Pods and
+create additional Pods as necessary.
+
+**TODO:** Non-cascading deletion as part of rollback sounds brittle. Replacing
```

Comment:
This is the real pain point of ending up with duplicate ReplicaSets (RSs that have the same PodTemplateSpec).